public/pages/CreateTrigger/containers/ConfigureTriggers/ConfigureTriggers.js

```diff
@@ -92,8 +87,18 @@ class ConfigureTriggers extends React.Component {
     const prevInputs = prevProps.monitor.inputs[0];
     const currInputs = this.props.monitor.inputs[0];
     if (!_.isEqual(prevInputs, currInputs)) {
-      const { isBucketLevelMonitor } = this.state;
-      if (isBucketLevelMonitor) this.onQueryMappings();
+      const {
+        monitor: { monitor_type },
+        monitorValues: { uri },
+      } = this.props;
+      switch (monitor_type) {
+        case MONITOR_TYPE.BUCKET_LEVEL:
+          this.onQueryMappings();
+          break;
+        case MONITOR_TYPE.CLUSTER_METRICS:
+          if (canExecuteClusterMetricsMonitor(uri)) this.onRunExecute();
+          break;
+      }
```

Comment:
Same as lines 47 to 58 above. Let's extract into a method

Reply:
Good catch. Consolidated this duplicate code.